content/en/real_user_monitoring/guide/understanding-the-rum-event-hierarchy.md

```diff
@@ -0,0 +1,112 @@
+
+#### Sample search: Top list of all "Add to cart" actions that led to an error 
+
+In this example, search within the actions event for an action error count (of at least 1) and specific action name to see how many times clicking "Add to cart" led to a user encountering an error.
+
+{{< img src="real_user_monitoring/guide/understanding-rum-event-hierarchy/rum-actions-all-add-to-cart.png" alt="Diagram of the RUM event hierarchy." style="width:80%;">}}
+
+### Errors
+
+You can use RUM to collect [frontend errors][6] that occur during the user session. By default, the browser SDK creates error events for unhandled exceptions and console errors. Additionally, you can collect custom errors through the RUM addError API ([browser][7] | [mobile][8]). On mobile apps, you can also see if the error led to a session termination, also known as a crash.
+
+Errors can be viewed across both RUM and Error Tracking. Source and custom errors will be processed by Error Tracking, while console errors will be solely in RUM.
+
+#### Sample search: List of all crashes that occurred on the "CheckoutViewController"page in the iOS Shop.ist application
+
+In this example, search within the errors event for the application ID, crashes, and view (page) name to see all crashes that occurred on the "CheckoutViewController" page for a particular operating system and application.
+
+{{< img src="real_user_monitoring/guide/understanding-rum-event-hierarchy/rum-sample-search-checkoutviewcontroller.png" alt="Diagram of the RUM event hierarchy." style="width:80%;">}}
+
+### Resources
+Resources are collected from views and include external requests from your application to a network provider, like XHR, JS loading, images, fonts, etc. Since it is collected from a view, you can query for all resources loaded on an application, or scope it down to just resources that occurred in a single view. 
+
+#### Sample search: A list of all images loaded on the /cart view, where the image size was greater than or equal to 1000 kilobytes (or 1MB)
+
+In this example, select **Resources** from the event type dropdown, then make a query for images that loaded on the cart view and were larger or equal to 1000 kilobytes.
+
+{{< img src="real_user_monitoring/guide/understanding-rum-event-hierarchy/rum-resources.png" alt="Diagram of the RUM event hierarchy." style="width:80%;">}}
+
+### Long tasks
+Long tasks are anything that blocks the UI thread for a given period of time. On mobile, for example, a long task may be a frozen frame if the screen is blocked for longer than 300 ms.
+
+#### Sample search: All frozen frame long tasks that lasted more than 500 ms
+
+In this example, select Long tasks from the event type dropdown and specify the duration.
+
+{{< img src="real_user_monitoring/guide/understanding-rum-event-hierarchy/rum-long-tasks.png" alt="Diagram of the RUM event hierarchy." style="width:80%;">}}
+
+## No data appears after writing a query
```

Comment:
i felt like going into this part of the guide as a bit abrupt -- can we add some extra spacing or a line of some sort (or even verbiage) to indicate we are now transitioning to a troubleshooting part of the guide?

Reply:
Good point. Added a troubleshooting header.